### PR TITLE
Purification

### PR DESCRIPTION
**Pull request by david-christiansen**
Idris and its libraries have been purified.

* Deprecate "Monad.return" in favor of "pure"
* Update all included libraries to use "pure"
* Update internal code generation to remove "return"
* Remove unused AST constructor PReturn
* Update tests

Also, a missing cast that caused warnings on GCC on the latest Debian in the RTS has been added.

### CODE REVIEW

**Comment by melted** [comment]
You added `Idris/Core/TT-i.hs`, that seems unrelated.

**Comment by melted** [comment]
Travis and Appveyor is complaining about failing to import System/Concurrency/Sessions.

**Comment by melted** [comment]
Another thing broke the build https://github.com/idris-lang/Idris-dev/pull/3382

File: test/meta002/AgdaStyleReflection.idr [hunk 1]
```diff
@@ -136,13 +136,13 @@ resolveVar k =
...
         getVar _     []         = fail [TextPart "Variable out of scope"]
-        getVar Z     (n :: _  ) = return n -- NB assumes unique names!
+        getVar Z     (n :: _  ) = pure n -- NB assumes unique names!
         getVar (S k) (_ :: env) = getVar k env
 
 ||| Prepare to apply a global by matching its argument plicities to
 ||| its argument list.
 prepareApply : List FunArg -> List (Arg Term) -> Elab (List (Arg (Maybe Term)))
-prepareApply [] tms = pure $ map (map Just) tms -- overapplied OK - it may return a function!
+prepareApply [] tms = pure $ map (map Just) tms -- overapplied OK - it may pure a function!
```
Comment (Melvar) [inline test/meta002/AgdaStyleReflection.idr]: This one should probably not be replaced.

**Comment by david-christiansen** [comment]
Thanks for catching my oversights, @melted and @Melvar, and for fixing the build, @melted!

**Comment by david-christiansen** [comment]
I'll re-run the CI just to make sure, then merge, unless an objection pops up in the meantime.

**Pull request merged**